### ISSUE TITLE
Crashes when used with @sanity/color-input

### ISSUE THREAD

**Issue by eghernqvist** [ndimatteo/sanity-plugin-note-field#1]
Got below error and scratched my head for a good 2 hours before realizing it was because of having installed the plugin [@sanity/color-input](https://github.com/sanity-io/sanity/tree/next/packages/%40sanity/color-input) which for some reason I cannot figure out clashes with using note-field, even though I am not even using a `type: "color"` input in the same schema, simply including the plugin in `sanity.json` leads to the desk crashing.

My usage of note-field, removing this from this `type` stops the form from crashing:
```js
{
  name: "note",
  type: "note",
  localize: false,
  options: {
    message:
      "Keep title short, as it will be shown in a very tiny space. Usually 1 word is preferred.",
    tone: "transparent"
  },
},
```

My `sanity.json`:
```json
{
  "root": true,
  "project": {
    "name": "..."
  },
  "api": {
    "projectId": "...",
    "dataset": "..."
  },
  "plugins": [
    "@sanity/base",
    "@sanity/components",
    "@sanity/default-layout",
    "@sanity/default-login",
    "@sanity/desk-tool",
    "@sanity/color-input",
    "studio-logo",
    "note-field",
    "create-logo",
    "default-preview",
    "translations"
  ],
  "env": {
    "development": {
      "plugins": [
        "@sanity/vision"
      ]
    }
  },
  "parts": [
    {
      "name": "part:@sanity/base/schema",
      "path": "./schemas/schema"
    },
    {
      "implements": "part:@sanity/base/theme/variables/override-style",
      "path": "variableOverrides.css"
    },
    {
      "name": "part:@sanity/desk-tool/structure",
      "path": "./deskStructure.js"
    }
  ]
}
```

```
Stack trace:
Error: Can only set one of `children` or `props.dangerouslySetInnerHTML`.
    at assertValidProps (/static/js/vendor.bundle.js:8118:15)
    at setInitialProperties (/static/js/vendor.bundle.js:9402:3)
    at finalizeInitialChildren (/static/js/vendor.bundle.js:10876:3)
    at completeWork (/static/js/vendor.bundle.js:22355:17)
    at completeUnitOfWork (/static/js/vendor.bundle.js:25569:16)
    at performUnitOfWork (/static/js/vendor.bundle.js:25542:12)
    at workLoopSync (/static/js/vendor.bundle.js:25507:22)
    at performSyncWorkOnRoot (/static/js/vendor.bundle.js:25133:9)
    at http://localhost:3333/static/js/vendor.bundle.js:14466:24
    at unstable_runWithPriority (/static/js/vendor.bundle.js:29135:12)
Component stack:
    in div (created by styled.div)
    in styled.div (created by ForwardRef(Text2))
    in ForwardRef(Text2) (created by ForwardRef)
    in div (created by ForwardRef(Inline2))
    in div (created by styled.div)
    in styled.div (created by ForwardRef(Box2))
    in ForwardRef(Box2) (created by Styled(Component))
    in Styled(Component) (created by ForwardRef(Inline2))
    in ForwardRef(Inline2) (created by ForwardRef)
    in div (created by styled.div)
    in styled.div (created by ForwardRef(Box2))
    in ForwardRef(Box2) (created by Styled(Component))
    in Styled(Component) (created by ForwardRef(Card2))
    in Fe (created by ThemeProvider)
    in ThemeProvider (created by ThemeColorProvider)
    in ThemeColorProvider (created by ForwardRef(Card2))
    in ForwardRef(Card2) (created by ForwardRef)
    in ForwardRef (created by FormBuilderInputInner)
    in ChangeIndicatorProvider (created by FormBuilderInputInner)
    in div (created by FormBuilderInputInner)
    in FormBuilderInputInner (created by FormBuilderInput)
    in FormBuilderInput (created by ForwardRef(ObjectInputField))
    in ConditionalField (created by ForwardRef(ObjectInputField))
    in ForwardRef(ObjectInputField) (created by ForwardRef(ObjectInput))
    in div (created by styled.div)
    in styled.div (created by ForwardRef(Box2))
    in ForwardRef(Box2) (created by Styled(Component))
    in Styled(Component) (created by ForwardRef(Grid2))
    in ForwardRef(Grid2) (created by ForwardRef(ObjectInput))
    in ForwardRef(ObjectInput)
    in ForwardRef(ObjectInput) (created by FormBuilderInputInner)
    in ChangeIndicatorProvider (created by FormBuilderInputInner)
    in div (created by FormBuilderInputInner)
    in FormBuilderInputInner (created by FormBuilderInput)
    in FormBuilderInput (created by SanityFormBuilder)
    in FormBuilderContext (created by SanityFormBuilderContext)
    in SanityFormBuilderContext (created by SanityFormBuilder)
    in SanityFormBuilder (created by EditForm)
    in form (created by EditForm)
    in EditForm (created by FormView)
    in div (created by ForwardRef(RegionsWithIntersections))
    in div (created by ForwardRef(RegionsWithIntersections))
    in ForwardRef(RegionsWithIntersections) (created by StickyOverlay)
    in StickyOverlay (created by OverlayEnabled)
    in Tracker (created by OverlayEnabled)
    in OverlayEnabled (created by FormView)
    in div (created by FormView)
    in FormView (created by DocumentPanel)
    in div (created by ForwardRef(ScrollContainer))
    in ForwardRef(ScrollContainer) (created by DocumentPanel)
    in div (created by DocumentPanel)
    in BoundaryElementProvider (created by DocumentPanel)
    in PortalProvider (created by DocumentPanel)
    in div (created by styled.div)
    in styled.div (created by ForwardRef(Box2))
    in ForwardRef(Box2) (created by Styled(Component))
    in Styled(Component) (created by ForwardRef(Card2))
    in Fe (created by ThemeProvider)
    in ThemeProvider (created by ThemeColorProvider)
    in ThemeColorProvider (created by ForwardRef(Card2))
    in ForwardRef(Card2) (created by DocumentPanel)
    in DocumentPanel (created by DocumentPane)
    in div (created by DocumentPane)
    in div (created by ForwardRef(ScrollContainer))
    in ForwardRef(ScrollContainer) (created by EnabledChangeConnectorRoot)
    in Tracker (created by EnabledChangeConnectorRoot)
    in EnabledChangeConnectorRoot (created by DocumentPane)
    in DialogProvider (created by DocumentPane)
    in div (created by KeyboardShortcutResponder)
    in KeyboardShortcutResponder (created by GetHookCollectionState)
    in GetHookCollectionState (created by RenderActionCollectionState)
    in RenderActionCollectionState (created by DocumentActionShortcuts)
    in DocumentActionShortcuts (created by DocumentPane)
    in LayerProvider (created by LegacyLayerProvider)
    in LegacyLayerProvider (created by DocumentPane)
    in DocumentPane (created by DocumentPaneProvider)
    in DocumentHistoryProvider (created by DocumentPaneProvider)
    in DocumentPaneProvider
    in StreamingComponent
    in StreamingComponent (created by Context.Consumer)
    in WithInitialValueWrapper (created by DeskToolPane)
    in DeskToolPane (created by DeskToolPanes)
    in SplitPaneWrapper (created by DeskToolPanes)
    in div (created by Pane)
    in Pane (created by SplitPane)
    in div (created by SplitPane)
    in SplitPane (created by PanesSplitController)
    in div (created by PanesSplitController)
    in div (created by Pane)
    in Pane (created by SplitPane)
    in div (created by SplitPane)
    in SplitPane (created by PanesSplitController)
    in div (created by PanesSplitController)
    in div (created by Pane)
    in Pane (created by SplitPane)
    in div (created by SplitPane)
    in SplitPane (created by PanesSplitController)
    in div (created by PanesSplitController)
    in div (created by Pane)
    in Pane (created by SplitPane)
    in div (created by SplitPane)
    in SplitPane (created by PanesSplitController)
    in div (created by PanesSplitController)
    in PanesSplitController (created by DeskToolPanes)
    in PortalProvider (created by DeskToolPanes)
    in div (created by styled.div)
    in styled.div (created by DeskToolPanes)
    in DeskToolPanes (created by DeskTool)
    in DeskTool (created by withRouter(DeskTool))
    in withRouter(DeskTool) (created by DeskToolPaneStateSyncer)
    in DeskToolPaneStateSyncer (created by DeskTool)
    in DeskToolFeaturesProvider (created by DeskTool)
    in DeskTool (created by RenderTool)
    in RenderTool (created by SchemaErrorReporter)
    in RouteScope (created by SchemaErrorReporter)
    in div (created by SchemaErrorReporter)
    in div (created by SchemaErrorReporter)
    in div (created by SchemaErrorReporter)
    in SchemaErrorReporter (created by DefaultLayout)
    in DefaultLayout (created by withRouter(DefaultLayout))
    in withRouter(DefaultLayout) (created by DefaultLayoutRoot)
    in RouterProvider (created by DefaultLayoutRoot)
    in LoginWrapper (created by DefaultLayoutRoot)
    in DefaultLayoutRoot (created by AppProvider)
    in div (created by styled.div)
    in styled.div (created by ForwardRef(Box2))
    in ForwardRef(Box2) (created by Styled(Component))
    in Styled(Component) (created by ForwardRef(Card2))
    in Fe (created by ThemeProvider)
    in ThemeProvider (created by ThemeColorProvider)
    in ThemeColorProvider (created by ForwardRef(Card2))
    in ForwardRef(Card2) (created by Styled(Component))
    in Styled(Component) (created by AppProvider)
    in SnackbarProvider (created by AppProvider)
    in ToastProvider (created by AppProvider)
    in LayerProvider (created by AppProvider)
    in PortalProvider (created by AppProvider)
    in UserColorManagerProvider (created by AppProvider)
    in AppProvider (created by SanityRoot)
    in Fe (created by ThemeProvider)
    in ThemeProvider (created by SanityRoot)
    in ZIndexProvider (created by SanityRoot)
    in SanityRoot
    in AppContainer
 ```

**Comment by ndimatteo** [comment]
Hey there @eghernqvist I don't think this is an issue with the Note Field plugin, or a conflict with the Color Input plugin.

In HULL, we use both of these fields without issue (https://github.com/ndimatteo/HULL/blob/main/studio/sanity.json#L13), which tells me this is some other issue in your setup.

The error your getting implies some component might be trying to set it's contents with `dangerouslySetInnerHTML` _and_ has `children`.

**Comment by GLips** [comment]
Hey @ndimatteo, we don't have the color input plugin installed on our version, but are also getting the same error message from the note plugin. It's just cropped up recently—was working fine before.

I'm digging a bit deeper into the issue to see if I can figure it out, but I'm pretty sure it's coming from this plugin.

**Comment by ndimatteo** [comment]
Hey there @GLips I looked into this further and I found that Sanity pushed some breaking UI updates with their `<Text />` component ([see here](https://github.com/sanity-io/design/blob/next/packages/%40sanity/ui/src/primitives/text/text.tsx#L78)).

I pushed up a fix in version `1.1.2` that should at least resolve this conflict causing studios to crash, but just a heads up: it seems there are now issues around using HTML in your Note message.

I'm working with the Sanity team to figure out a solution there, but for basic note messages you should be all set 🤘

**Comment by ndimatteo** [comment]
Alright @GLips and @eghernqvist, I got this sorted!

You'll want to be on the latest studio versions: `^2.19.0` (this includes the new `styled-components` depedency)

That, plus the update here (`1.1.2`) will allow you to use Note fields without issue.

One notable change is if you plan to include HTML in your note `message` you will need to rewrite these from strings to JSX. I've updated the [second example](https://github.com/ndimatteo/sanity-plugin-note-field#options) in the readme with how to do this.
